File: works/6588.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main
{
    static final int MAX = 1000001;
    static int[] primeNumber = new int[MAX];

    static void FindPrimeNumbers()
    {
        for(int i = 2; i < MAX; i++)
        {
            primeNumber[i] = i;
        }

        for(int i = 2; i < Math.sqrt(MAX); i++)
        {
            if(primeNumber[i] == 0) continue;
            for(int j = i * i; j < MAX; j += i)
            {
                primeNumber[j] = 0;
            }
        }
    }

    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        FindPrimeNumbers();

        while(true)
        {
            int n = Integer.parseInt(br.readLine());
            if(n == 0) break;
            int a = -1;
            int b = -1;
            for(int i = 1; i < n; i+=2)
            {
                if(primeNumber[i] != 0 && primeNumber[n-i] != 0)
                {
                    a = i;
                    b = n-i;
                    break;
                }
            }
            if(a == -1 && b == -1)
                System.out.println("Goldbach's conjecture is wrong.");
            else
                System.out.println(n + " = " + a + " + " + b);
        }

    }

}

File: works/js/3-8/6588.js
let fs = require('fs');
let input = fs.readFileSync(0).toString().trim().split('\n');
// let input = fs.readFileSync('input.txt').toString().split('\n');

let MAX = 1000001
let primeNumber = Array(MAX) 
let answer = "" 

function FindPrimeNumber() {
    primeNumber[1] = 0 
    for(i = 2; i < MAX; i++) {
        primeNumber[i] = i 
    }
    for(i = 2; i*i < MAX; i++) {
        if(primeNumber[i] == 0) continue;
        for(j = i*i; j < MAX; j += i) {
            primeNumber[j] = 0 
        }
    }
}

FindPrimeNumber() 


for(tc = 0; tc < input.length-1; tc++) {
    let n = parseInt(input[tc]) 
    let a = -1, b = -1 
    
    for(i = 1; i < n; i += 2) {
        if(primeNumber[i] != 0 && primeNumber[n-i] != 0) {
            a = i 
            b = n-i 
            break 
        }
    }

    if(a == -1 && b == -1) {
        answer += "Goldbach's conjecture is wrong.\n"
    }
    else {
        answer += n + " = " + a + " + " + b + "\n"
    }
}

console.log(answer)